File: frontend/src/model/loyalty.ts
/** Models for table: loyalty (ids are number / BIGINT) */

export class LoyaltyRes {
  id: number = 0;
  orgId: number = 0;
  locationId: number = 0;
  name: string = "";
  type: string = "";
  earnRate: string = "";
  redeemValue: string = "";
  tier: string = "";
  minSpend: number = 0;
  expiryMonths: number = 0;
  qrEnabled: string = "";
  status: string = "";
  createdby: string = "";
  createdon: string = "";
  updatedby: string = "";
  updatedon: string = "";
  earnUnitRupees: number = 0;
  pointsPerUnit: number = 0;
  rupeesPerPoint: number = 0;
  rewardDescription: string = "";
  stampsRequired: number = 0;
  spinsAllowed: string = "";
  requiresCheckIn: string = "";
}

export class CreateLoyaltyReq {
  orgId: number = 0;
  locationId: number = 0;
  name: string = "";
  type: string = "";
  earnRate: string = "";
  redeemValue: string = "";
  tier: string = "";
  minSpend: number = 0;
  expiryMonths: number = 0;
  qrEnabled: string = "";
  status: string = "";
  earnUnitRupees: number = 0;
  pointsPerUnit: number = 0;
  rupeesPerPoint: number = 0;
  rewardDescription: string = "";
  stampsRequired: number = 0;
  spinsAllowed: string = "";
  requiresCheckIn: string = "";
}

export class CreateLoyaltyRes extends LoyaltyRes {}

export class UpdateLoyaltyReq {
  id: number = 0;
  orgId: number = 0;
  locationId: number = 0;
  name: string = "";
  type: string = "";
  earnRate: string = "";
  redeemValue: string = "";
  tier: string = "";
  minSpend: number = 0;
  expiryMonths: number = 0;
  qrEnabled: string = "";
  status: string = "";
  earnUnitRupees: number = 0;
  pointsPerUnit: number = 0;
  rupeesPerPoint: number = 0;
  rewardDescription: string = "";
  stampsRequired: number = 0;
  spinsAllowed: string = "";
  requiresCheckIn: string = "";
}

export class UpdateLoyaltyRes extends LoyaltyRes {}

export class GetLoyaltyReq {
  id: number = 0;
  orgId: number = 0;
}

export class GetLoyaltyRes extends LoyaltyRes {
  errorMessage: string = "";
}

export class ListLoyaltyReq {
  orgId: number = 0;
  locationId: number = 0;
  search: string = "";
  page: number = 1;
  pageSize: number = 50;
}

export class ListLoyaltyRes {
  items: LoyaltyRes[] = [];
  total: number = 0;
  errorMessage: string = "";
}

export class DeleteLoyaltyReq {
  id: number = 0;
  orgId: number = 0;
}

export class DeleteLoyaltyRes {
  id: number = 0;
  success: boolean = false;
  errorMessage: string = "";
}

File: backend/Krios/krios/Models/Loyalty.cs
namespace Krios.Models.Krios
{
    public class Loyalty
    {
        public long id { get; set; }
        public long orgId { get; set; }
        public long locationId { get; set; }
        public string name { get; set; } = "";
        public string type { get; set; } = "";
        public string earnRate { get; set; } = "";
        public string redeemValue { get; set; } = "";
        public string tier { get; set; } = "";
        public decimal minSpend { get; set; }
        public long expiryMonths { get; set; }
        public string qrEnabled { get; set; } = "";
        public string status { get; set; } = "";
        public string createdby { get; set; } = "";
        public DateTime? createdon { get; set; }
        public string updatedby { get; set; } = "";
        public DateTime? updatedon { get; set; }
        public decimal earnUnitRupees { get; set; }
        public decimal pointsPerUnit { get; set; }
        public decimal rupeesPerPoint { get; set; }
        public string rewardDescription { get; set; } = "";
        public long stampsRequired { get; set; }
        public string spinsAllowed { get; set; } = "";
        public string requiresCheckIn { get; set; } = "";
    }

    public class LoyaltySelectReq
    {
        public long id { get; set; }
        public long orgId { get; set; }
        public long locationId { get; set; }
        public string search { get; set; } = "";
        public string status { get; set; } = "";
    }

    public class LoyaltyDeleteReq
    {
        public long id { get; set; }
        public long orgId { get; set; }
    }
}